x/gamm/pool-models/stableswap/pool.go

```diff
@@ -0,0 +1,86 @@
+package stableswap
+
+import (
+	"encoding/json"
+	fmt "fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/types"
+)
+
+var _ types.PoolI = &Pool{}
+
+func (pa Pool) GetAddress() sdk.AccAddress {
+	addr, err := sdk.AccAddressFromBech32(pa.Address)
+	if err != nil {
+		panic(fmt.Sprintf("could not bech32 decode address of pool with id: %d", pa.GetId()))
+	}
+	return addr
+}
+
+func (pa Pool) String() string {
+	out, err := json.Marshal(pa)
+	if err != nil {
+		panic(err)
+	}
+	return string(out)
+}
+
+func (pa Pool) GetId() uint64 {
+	return pa.Id
+}
+
+func (pa Pool) GetSwapFee(ctx sdk.Context) sdk.Dec {
+	return pa.PoolParams.SwapFee
+}
+func (pa Pool) GetExitFee(ctx sdk.Context) sdk.Dec {
+	return pa.PoolParams.ExitFee
+}
+
+func (pa Pool) IsActive(ctx sdk.Context) bool {
+	return true
+}
+
+// Returns the coins in the pool owned by all LP shareholders
+func (pa Pool) GetTotalPoolLiquidity(ctx sdk.Context) sdk.Coins {
+	return pa.PoolLiquidity
+}
+
+func (pa Pool) GetTotalShares() sdk.Int {
+	return pa.TotalShares.Amount
+}
+
+// These should all get moved to amm.go
+func (pa Pool) CalcOutAmtGivenIn(ctx sdk.Context, tokenIn sdk.Coins, tokenOutDenom string, swapFee sdk.Dec) (tokenOut sdk.DecCoin, err error) {
+	return sdk.DecCoin{}, types.ErrNotImplemented
+}
+func (pa *Pool) SwapOutAmtGivenIn(ctx sdk.Context, tokenIn sdk.Coins, tokenOutDenom string, swapFee sdk.Dec) (tokenOut sdk.Coin, err error) {
```

Comment:
Asking to learn:

I noticed that in the `balancer` package some of the methods that define `PoolI` are in `amm.go` and others are in `balancer_pool.go`.

Is there a reason for that?

Would it make sense to rename `balancer/balancer_pool.go` to `balancer/pool.go` so that the pool implementations are easier to navigate due to the consistency in naming across `pool-models`

Reply:
Agree with @p0mvn here. A single file `pool.go` makes sense to me.

Reply:
I think a single `pool.go` file makes sense.

I do think having all the math files in one location is preferrable though. If theres boilerplate wrapper / error checking, we could do that in pool.go and then call into a private function in `amm.go`